crates/torii/core/src/utils.rs

```diff
@@ -46,6 +48,18 @@ pub async fn fetch_content_from_ipfs(cid: &str, mut retries: u8) -> Result<Bytes
         IPFS_CLIENT_MAX_RETRY, cid
     )))
 }
+
+pub async fn health_check_provider<P: Provider + Sync + std::fmt::Debug + 'static>(
+    provider: Arc<P>,
+) {
+    let latest_block = provider.get_block_with_tx_hashes(BlockId::Tag(BlockTag::Latest)).await;
+    if let Ok(latest_block) = latest_block {
+        info!("Provider health_check latest_block is: {:?}", latest_block);
+    } else {
+        error!("Provider: {:?} is unhealthy. please check you config!", provider);
```

Comment:
```suggestion
        error!(?provider, "Unhealthy provider, please check your configuration.", provider);
```
Use structure logging instead.

Reply:
I think for now the provider doesn't expose the URL by default, so we may keep the provider printed as default for now. In the future we would rather use something like `provider.url()` if added in starknet-rs. 👍

Reply:
fixed.